### PR TITLE
HGDP + 1KG join call stats for update

### PR DESCRIPTION
**Pull request by jkgoodrich**
Functions copied from this PR: https://github.com/broadinstitute/gnomad_qc/pull/390

This contains the functions required to join the call stats from the v3.1.2 sites HT to the call stats for the updated samples into one HT.

The first commit is basically a copy from the above PR, and any commits after that are my suggested changes.

### CODE REVIEW

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 1]
```diff
@@ -707,11 +819,17 @@ def main(args):
     v4_genome_release_resources = get_v4_genomes_release_resources(
         test=test, overwrite=overwrite
     )
-    meta_ht = v4_genome_release_resources.meta_ht.ht()
-    dense_mt = v4_genome_release_resources.dense_mt.mt()
+    v3_hgdp_tgp_meta_ht = v4_genome_release_resources.meta_ht.ht()
+    v3_hgdp_tgp_dense_mt = v4_genome_release_resources.dense_mt.mt()
+    v3_hgdp_tgp_sites_ht = v4_genome_release_resources.sites_ht.ht()
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:824]: I think the name here is again a bit confusing, can we use `v3_sites_ht` or `v3_release_sites_ht` instead?

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 2]
```diff
@@ -707,11 +819,17 @@ def main(args):
...
 
     if test:
-        dense_mt = filter_to_test(dense_mt, gene_on_chrx=gene_on_chrx)
+        v3_hgdp_tgp_dense_mt = filter_to_test(
+            v3_hgdp_tgp_dense_mt, gene_on_chrx=gene_on_chrx
+        )
+        v3_hgdp_tgp_sites_ht = filter_to_test(
+            v3_hgdp_tgp_sites_ht, gene_on_chrx=gene_on_chrx
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:831]: if you are going to change the name, here too.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 3]
```diff
@@ -755,6 +873,15 @@ def main(args):
 
             freq_ht.write(getattr(res, f"freq_{name}_ht").path, overwrite=overwrite)
 
+    if args.join_callstats_for_update:
+        res = v4_genome_release_resources.join_callstats_for_update
+        res.check_resource_existence()
+        logger.info("Joining all callstats HTs with the release sites HT...")
+        join_release_ht_with_subsets(
+            v3_hgdp_tgp_sites_ht,
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:881]: here too!

**Review by KoalaQin — CHANGES_REQUESTED** [review]
Only some naming comments if you agree.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py (hunk 1 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:824]: Oops, yup, I didn't mean to change that one, thanks for catching

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 4]
```diff
@@ -707,11 +822,15 @@ def main(args):
     v4_genome_release_resources = get_v4_genomes_release_resources(
         test=test, overwrite=overwrite
     )
-    meta_ht = v4_genome_release_resources.meta_ht.ht()
-    dense_mt = v4_genome_release_resources.dense_mt.mt()
+    v3_hgdp_tgp_meta_ht = v4_genome_release_resources.meta_ht.ht()
+    v3_hgdp_tgp_dense_mt = v4_genome_release_resources.dense_mt.mt()
+    v3_sites_ht = v4_genome_release_resources.sites_ht.ht()
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:827]:
```suggestion
    v3_sites_ht = v4_genome_release_resources.v3_sites_ht.ht()
```
You need a minor change here to make it work.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 5]
```diff
@@ -590,6 +609,87 @@ def filter_freq_arrays(
...
+    freq_hts = {"release": ht, **freq_hts}
+    for name, freq_ht in freq_hts.items():
+        logger.info("There are %i variants in the %s HT...", freq_ht.count(), name)
+        freq_hts[name] = freq_ht.select("freq").select_globals(*FREQ_GLOBALS)
+
+    logger.info("Joining release HT with AFs for added and subtracted samples...")
+    freq_hts = [freq_hts[x] for x in JOIN_FREQS]
+    ht = hl.Table.multi_way_zip_join(freq_hts, "ann_array", "global_array")
+
+    logger.info("Filtering joint HT to rows where at least one group has AC > 0...")
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:687]: Okay, so we filter here first?

**Review by KoalaQin — CHANGES_REQUESTED** [review]
Two more small comments.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py (hunk 4 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:827]: thank you!

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py (hunk 5 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:687]: Yeah, I don't think there is any reason to store variants in this table that are not going to be in the release

**Comment by KoalaQin** [comment]
Tested. LGTM!

**Review by KoalaQin — APPROVED** [review]

**Pull request merged**